### DISCUSSION TITLE
First Time using Python and Pip

### DISCUSSION THREAD

**Discussion by WanderingVagabond** [AnthonyMichaelTDM/DRG-Save-Editor#9] [Q&A] [answered]
Greetings! I would like to inquire if it would be possible to make a video/tutorial around how to implement the pip/python requirements to obtain the S2 Overclocks?

**Comment by AnthonyMichaelTDM** [comment]
Hmm.
I'm going to be gone this weekend but I could probably throw something together after.

**Reply by AnthonyMichaelTDM** [reply]
When I get around to that, I'll also remove the obsolete .zip file

**Reply by WanderingVagabond** [reply]
Thanks, I'm very new to Python/Pip and can't really make heads-or-tales about it. Most likely just me being shit at programming stuff.

**Comment by AnthonyMichaelTDM** [comment, marked as answer]
I'd like to thank @NerdyJosh1 for helping us all out and making a tutorial video, I've reviewed it and it touches all the bases, I'm going to drop a link below and mark this issue as answered
#12 

https://www.youtube.com/watch?v=2h2-nZ2ptRo&ab_channel=NerdyJosh

**Reply by WanderingVagabond** [reply]
Whilst the tutorial certainly helped, still getting an error that I'm not sure why since I followed the video.

Desktop\DRG-Save-Editor-master> pip install -r ./requirements.txt
pip : The term 'pip' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the
spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ pip install -r ./requirements.txt
+ ~~~
    + CategoryInfo          : ObjectNotFound: (pip:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

**Reply by AnthonyMichaelTDM** [reply]
@WanderingVagabond 
You may need to install pip, what operating system are you using and where did you install Python?

**Reply by WanderingVagabond** [reply]
I simply followed as the video asked, so Python came from Windows Store. As for my OS, it's Windows 10 Pro.

**Reply by WanderingVagabond** [reply]
Also, doesn't Pip already come alongside the Python install?

**Reply by WanderingVagabond** [reply]
Any ideas on what I'm doing wrong?

**Reply by AnthonyMichaelTDM** [reply]
> Also, doesn't Pip already come alongside the Python install?

Yes, that's why I asked.
The error message indicates that pip either isn't installed, or that where it's installed isn't in your PATH environment variable

**Reply by AnthonyMichaelTDM** [reply]
> Any ideas on what I'm doing wrong?

What terminal are you using?
Powershell or Command Prompt (cmd.exe)

**Reply by WanderingVagabond** [reply]
I'm using Powershell as the tutorial depicted. How do I install Pip with the Windows Store Python? Or correct the PATH variable?

**Reply by AnthonyMichaelTDM** [reply]
> I'm using Powershell as the tutorial depicted. How do I install Pip with the Windows Store Python? Or correct the PATH variable?

did you have PowerShell open before downloading python? if so restart PowerShell and try again.

**Reply by AnthonyMichaelTDM** [reply]
after trying that, if the issue persists try following this guide https://www.geeksforgeeks.org/how-to-install-pip-on-windows/ 

let me know if any of these things work for you :)

**Reply by WanderingVagabond** [reply]
I sincerely have zero idea what I'm doing wrong, I followed both the guide and the tutorial but I'm still getting the same error.

**Reply by WanderingVagabond** [reply]
I currently want to cry in both pain and laughter, decided to watch an installation guide for Python and noticed I missed one major thing during it. Which makes me feel like a complete idiot. But I think I can get it working now. Thank you for entertaining my gripping.

**Reply by WanderingVagabond** [reply]
Wait, got it to install. Though new problem on my end.

python3 ./src/main/python/main.py
python3 : The term 'python3' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ python3 ./src/main/python/main.py
+ ~~~~~~~
    + CategoryInfo          : ObjectNotFound: (python3:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

**Reply by WanderingVagabond** [reply]
nvm, got it to work.

python3 ./src/main/python/main.py

just had to make it python ./src/main/python/main.py

**Reply by AnthonyMichaelTDM** [reply]
Awesome, glad to hear it!

**Comment by cock-roach69** [comment]
Hey, I have the same issue as **@WanderingVagabond** had, but I don't know how to resolve it, I already tried reinstalling Python and running the 'python ./src/main/python/main.py' command.

python3 : The term 'python3' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ python3 ./src/main/python/main.py
+ ~~~~~~~
    + CategoryInfo          : ObjectNotFound: (python3:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

**Reply by AnthonyMichaelTDM** [reply]
Try `python` instead of `python3`

**Reply by cock-roach69** [reply]
I tried doing that but it didn't change anything, I get the same error message.

python : The term 'python' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ python ./src/main/python/main.py
+ ~~~~~~
    + CategoryInfo          : ObjectNotFound: (python:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
